app/src/lib/tailer.ts

```diff
@@ -43,6 +43,10 @@ export class Tailer {
 
     try {
       const watcher = Fs.watch(this.path, this.onWatchEvent)
+      watcher.on('error', error => {
```

Comment:
Does an error automatically close the watcher or should we make sure that it's closed?

Also, should we expose this error through some event for consumers? Seems like they'll never know that tailing has failed?

Reply:
> Does an error automatically close the watcher or should we make sure that it's closed?

I believe so. Manually closing the FSWatcher closes the underlying handle (of type `FSEvent`).

```js
FSWatcher.prototype.close = function() {
  this._handle.close();
};
```

Inside `onchange` it'll do the same:

```js
  this._handle.onchange = function(status, eventType, filename) {
    if (status < 0) {
      self._handle.close();
      ...
      self.emit('error', error);
    } else {
      self.emit('change', eventType, filename);
    }
  };
```

> Also, should we expose this error through some event for consumers? Seems like they'll never know that tailing has failed?

I haven't thought much about what consumers would want to do to handle an `error` event being emitted. This is currently only use for parsing Git LFS progress, and an error (either when initializing the file or when `onchange` occurs) means that we "give up" on reporting progress.

If there's something UI-wise that we should be doing here to report the lack of progress, then I'm :thumsbup: to emitting this. But I wouldn't hold up this fix to the unhandled error unless you think it's important.

Reply:
> I believe so. Manually closing the FSWatcher closes the underlying handle (of type FSEvent).

Excellent!

> This is currently only use for parsing Git LFS progress, and an error (either when initializing the file or when onchange occurs) means that we "give up" on reporting progress and then the tailer is disposed.

> I wouldn't hold up this fix to the unhandled error unless you think it's important.

I concur that it has little practical application in our use case at the moment but I think it makes conceptual sense to have the Tailer class expose errors. The decision about whether or not to care about not being able to tail a file should fall on the consumer, not be hidden away in the implementation. Similarly I think we should capture and propagate the error event from the read stream and let the LFS consumer handle that as well.

Since we're not doing a release this week I think it makes sense to address this while we're in here.